Tools/CISupport/ews-build/steps.py

```diff
@@ -1463,7 +1463,33 @@ class UnApplyPatchIfRequired(CleanWorkingDirectory):
     descriptionDone = ['Unapplied patch']
 
     def doStepIf(self, step):
-        return self.getProperty('patchFailedToBuild') or self.getProperty('patchFailedTests')
+        return self.getProperty('patch_id') and (self.getProperty('changeFailedToBuild') or self.getProperty('changeFailedTests'))
+
+    def hideStepIf(self, results, step):
+        return not self.doStepIf(step)
+
+
+class RevertPullRequestIfRequired(steps.ShellSequence):
```

Comment:
I know 'IfRequired' word is inspired from existing step `UnApplyPatchIfRequired`, but I wanted to rename that to `UnApplyPatch`. Let's keep this one `RevertPullRequest`. Also we don't really need the properties `changeFailedToBuild` and `changeFailedTests`. 

Those were added in very early days of this instance bringup, when that step was listed statically in factories.py. Later on we started invoking steps dynamically using `addStepsAfterCurrentStep` and `UnApplyPatchIfRequired` is no longer called in factories.py. We can just remove all this `patchFailedToBuild` property logic. I can do that in a separate patch or you can do it if you want.

Reply:
Removing these properties in https://github.com/WebKit/WebKit/pull/111